addons/AMO-data-request.kp/knowledge.md

```diff
@@ -0,0 +1,490 @@
+---
+title: Query AMO with Add-on GUID
+authors:
+- Ben Miroglio
+tags:
+- AMO
+- add-ons
+- firefox-desktop
+created_at: 2017-01-09 00:00:00
+updated_at: 2017-01-09 13:34:04.550788
+tldr: Get metadata for an add-on through AMO given its GUID
+---
+# Query AMO with Add-on GUID
+
+In Telemetry and elsewhere we typically use add-on GUIDs to uniquely represent specific add-ons. Often times a GUID is ambiguous, revealing little to no information about the add-on. This script allows a user to quickly get add-on names, e10s compatibility, versions, weekly downloads, etc. from AMO with just an add-on GUID. See the Appendix for an example JSON blob displaying all possible fields. Aside from easily acquiring meta data for add-ons, this example shows the various fields the user can access *not* accessible via telemetry at the moment.
+
+The example below is a simplification of the [script](https://github.com/andymckay/new-arewee10syet.com/blob/master/build.py) used to generate the [arewee10syet.com](https://arewee10syet.com) page.
+
+
+
+```python
+from __future__ import division
+import pandas as pd
+import os
+import requests
+import json
+import urllib
+import sys
+```
+# Set Up
+
+
+
+```python
+# for manual editting of missing or incorrect add-on names
+fixups = {
+    'testpilot@labs.mozilla.com': 'Test Pilot (old one)',
+    '{20a82645-c095-46ed-80e3-08825760534b}': 'Microsoft .NET framework assistant',
+}
+
+def process_amo(result, compat):
+    """
+    reorganize AMO meta data and e10s compatibility into
+    single dict and return
+    """
+    try:
+        name = result['name']['en-US']
+    except KeyError:
+        name = result['slug']
+    return {
+        'name': name,
+        'url': result['url'],
+        'guid': result['guid'],
+        'status': compat['e10s'],
+        'id': result['id'],
+        'weekly_downloads': result['weekly_downloads'],
+        'ratings': result['ratings']
+    }
+
+def amo(guid, raw=False):
+    """
+    Make AMO API call to request data for a given add-on guid 
+    
+    Return raw data if raw=True, which returns a list of two dicts
+    [e10s_compatibility, AMO_meta_data]
+    
+    Otherwise call process_amo() to only return fields of interest 
+    (specified in process_amo())
+    """
+    addon_url = AMO_SERVER + '/api/v3/addons/addon/{}/'.format(guid)
+    compat_url = AMO_SERVER + '/api/v3/addons/addon/{}/feature_compatibility/'.format(guid)
+
+    data = []
+    for url in (addon_url, compat_url):
+        name = url.split('/')[-2]
+        if name != 'feature_compatibility':
```

Comment:
This block is a bit awkward. I think you could accomplish this by printing the status update with the GUID right before entering the `for` loop.